### PR TITLE
Avoid Full Path Disclosure error on session error.

### PR DESCRIPTION
**Pull request by ArthurHoaro**
* Add a function to validate session ID.
  * Generate a new session ID if an invalid token is passed.

related to #298 

ping @TeamAlexandriZ

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -82,6 +69,23 @@
     exit;
 }
 
+// Force cookie path (but do not change lifetime)
```
Comment (virtualtam) [inline index.php]: could this code section be cleaned up (indentation + comments)?

**Comment by virtualtam** [comment]
@Knah-Tsaeb @TeamAlexandriZ could you test this PR and give us some feedback? :)

**Comment by TeamAlexandriZ** [comment]
Give me a patched online target and I'll looking around.

File: index.php (hunk 1 above)
Comment (ArthurHoaro) [inline index.php]: Done.

**Comment by ArthurHoaro** [comment]
@TeamAlexandriZ Here you go: https://workspace.hoa.ro/shaarli-fpd/

**Comment by TeamAlexandriZ** [comment]
Hi, it's OK for cookie but on this server all php file of the folder /tests give a full path disclosure to :
https://workspace.hoa.ro/shaarli-fpd/tests/CachedPageTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/CacheTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/ConfigTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/LinkDBTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/TimeZoneTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/UrlTest.php
https://workspace.hoa.ro/shaarli-fpd/tests/UtilsTest.php

Error found : 
Warning: require_once(application/Utils.php): failed to open stream: No such file or directory in /var/www/html/codiad/workspace/shaarli-fpd/tests/UtilsTest.php on line 6

Fatal error: require_once(): Failed opening required 'application/Utils.php' (include_path='.:/usr/share/pear:/usr/share/php') in /var/www/html/codiad/workspace/shaarli-fpd/tests/UtilsTest.php on line 6

**Comment by virtualtam** [comment]
@TeamAlexandriZ thanks for testing!

@ArthurHoaro that's weird, there should be a `.htaccess` for the `tests/` directory preventing access

**Comment by ArthurHoaro** [comment]
I need to check but it might only be because I disallowed htaccess rules.

**Comment by ArthurHoaro** [comment]
Confirmed, with htaccess rules, those files return an error 403.

**Comment by virtualtam** [comment]
Merge-ready then ;-)

**Comment by TeamAlexandriZ** [comment]
On Nginx, the .htaccess doesn't work.

**Comment by virtualtam** [comment]
@TeamAlexandriZ Nginx's configuration and philosophy are quite different from Apache's (and related to server configuration rather than to this particular PR)

There is a wiki page dedicated to [Server configuration](https://github.com/shaarli/Shaarli/wiki/Server-configuration) which has sections for Apache2 and Nginx, and aims at providing Shaarli users with a sane, secure server setup -all improvements are welcome!

**Comment by TeamAlexandriZ** [comment]
Yes, but add 

location /tests {
  deny all;
  return 404;
}

to your /etc/nginx/deny.conf wiki example beause actually there are nothing who deny the access for Nginx users and the FUD is visible.

**Pull request merged**